### PR TITLE
Feature/android sms optimization

### PR DESCRIPTION
**Pull request by drilonreqica**
Optimized SmsConfig process for future when app will be used from jCenter and MavenCentral

### CODE REVIEW

File: secure-mobile-push/smp-android-client/app/src/main/java/de/adorsys/android/securemobilepushtest/MainActivity.java [hunk 1]
```diff
@@ -62,6 +62,7 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        SmsTool.createSmsConfig("BEGIN-MESSAGE", "END-MESSAGE", "0900123456", "0900654321", "0900900900");
```
Comment (itsmortoncornelius) [inline secure-mobile-push/smp-android-client/app/src/main/java/de/adorsys/android/securemobilepushtest/MainActivity.java]: see slack - use a singleton instead

File: secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsConfig.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package de.adorsys.android.securemobilepush.sms;
+
+import android.support.annotation.NonNull;
+
+class SmsConfig {
+    @NonNull
```
Comment (itsmortoncornelius) [inline secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsConfig.java]: see slack - use a singleton instead

File: secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsReceiver.java [hunk 3]
```diff
@@ -26,30 +27,39 @@
...
             Bundle bundle = intent.getExtras();
             SmsMessage[] smsMessages;
-            String messageFrom;
+            String messageFrom = null;
             if (bundle != null) {
                 try {
                     //PDU = protocol data unit
                     //A PDU is a “protocol data unit”, which is the industry format for an SMS message.
-                    //Because SMSMessage reads/writes them you shouldn’t need to disect them.
+                    //Because SMSMessage reads/writes them you should'nt need to dissect them.
```
Comment (itsmortoncornelius) [inline secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsReceiver.java]: should'nt --> shouldn't

File: secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsReceiver.java [hunk 4]
```diff
@@ -26,30 +27,39 @@
...
+                        // in to chunks of 153 characters before being received on the device.
+                        // To rectify that receivedMessage is the result of appending every single
+                        // short message into one large one for our usage. see:
+                        //http://www.textanywhere.net/faq/is-there-a-maximum-sms-message-length
+                        String receivedMessage = "";
                         for (int i = 0; i < smsMessages.length; i++) {
                             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                                 smsMessages[i] = SmsMessage.createFromPdu((byte[]) pdus[i],
                                         bundle.getString("format"));
+                                receivedMessage = receivedMessage + smsMessages[i].getMessageBody();
```
Comment (itsmortoncornelius) [inline secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsReceiver.java]: can be outside the if / else block as twice the same code

File: secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsTool.java [hunk 5]
```diff
@@ -21,4 +21,12 @@ public static void requestSMSPermission(@NonNull Activity activity) {
                     REQUEST_CODE_ASK_PERMISSIONS);
         }
     }
+
+    public static void createSmsConfig(@NonNull String BEGIN_INDEX,
```
Comment (itsmortoncornelius) [inline secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsTool.java]: can be removed --> directly use singleton

File: secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsReceiver.java (hunk 3 above)
Comment (drilonreqica) [inline secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsReceiver.java]: Fixed in commit: https://github.com/adorsys/secure-key-storage/pull/44/commits/e66adc11b4a7a93595cda41a2e21b2e3ffd3fc85

File: secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsReceiver.java (hunk 4 above)
Comment (drilonreqica) [inline secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsReceiver.java]: Fixed in commit: https://github.com/adorsys/secure-key-storage/pull/44/commits/e66adc11b4a7a93595cda41a2e21b2e3ffd3fc85

File: secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsTool.java (hunk 5 above)
Comment (drilonreqica) [inline secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsTool.java]: Fixed in commit:https://github.com/adorsys/secure-key-storage/pull/44/commits/eb220e2d2982822d17e2173262b0ad7e0fb3e81c

File: secure-mobile-push/smp-android-client/app/src/main/java/de/adorsys/android/securemobilepushtest/MainActivity.java (hunk 1 above)
Comment (drilonreqica) [inline secure-mobile-push/smp-android-client/app/src/main/java/de/adorsys/android/securemobilepushtest/MainActivity.java]: Fixed in commit:https://github.com/adorsys/secure-key-storage/pull/44/commits/eb220e2d2982822d17e2173262b0ad7e0fb3e81c

File: secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsConfig.java (hunk 2 above)
Comment (drilonreqica) [inline secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsConfig.java]: Fixed in commit:https://github.com/adorsys/secure-key-storage/pull/44/commits/eb220e2d2982822d17e2173262b0ad7e0fb3e81c

**Pull request merged**